src/ff8_opengl.cpp

```diff
@@ -505,6 +505,105 @@ LPDIJOYSTATE2 ff8_update_gamepad_status()
 	return ff8_externals.dinput_gamepad_state;
 }
 
+int ff8_get_input_device_capabilities_number_of_buttons(int a1)
+{
+	return xinput_connected ? 16 : joystick.GetCaps()->dwButtons;
```

Comment:
Since the memory allocation is only for 16 buttons, wouldn't it make sense to cap to 16 this capability anyway? Or at least check if `dwButtons < 16 ? dwButtons : 16` or something like this.

Reply:
The reality is worse: the game would never save more than 10 buttons in its mapping anyway